packages/login-ui/redux/thunks.js

```diff
@@ -127,6 +125,8 @@ export const onSubmitLoginForm = (password) => async (dispatch, getState) => {
         const defaultURL = await dispatch(SessionThunks.login(selectedAccount))
 
         dispatch(replace(localStorage.getLastURL() || defaultURL))
+        const signer = getEthereumSigner(getState())
```

Comment:
There is a lot of duplication in `WALLET_TYPE_MEMORY` and the alternative branches. Many lines of code are like two peas in a pod. It s required to move the duplicated code to reusable functions.

Reply:
I created a bug task for it (MINT-2087)